src/Tribe/Integrations/WPML/Filters.php

```diff
@@ -68,26 +68,40 @@ public function filter_tribe_events_rewrite_i18n_slugs_raw( $bases, $method, $do
 		$post_slug_translation_on  = ! empty( $sitepress_settings['posts_slug_translation']['on'] );
 
 		if ( $string_translation_active && $post_slug_translation_on ) {
-			$supported_post_types = array( Tribe__Events__Main::POSTTYPE );
+			$bases  = $this->translate_single_slugs( $bases );
+		}
 
-			foreach ( $supported_post_types as $post_type ) {
-				// check that translations are active for this CPT
-				$cpt_slug_is_not_translated = empty( $sitepress_settings['posts_slug_translation']['types'][ $post_type ] );
+		return $bases;
+	}
 
-				if ( $cpt_slug_is_not_translated ) {
-					continue;
-				}
+	/**
+	 * @param $bases
+	 *
+	 * @return array
+	 */
+	protected function translate_single_slugs( array $bases ) {
+		global $sitepress_settings;
 
-				$slug_translations = WPML_Slug_Translation::get_translations( $post_type );
+		$supported_post_types = array( Tribe__Events__Main::POSTTYPE );
 
-				if ( ! ( is_array( $slug_translations ) && isset( $slug_translations[1] ) ) ) {
-					continue;
-				}
+		foreach ( $supported_post_types as $post_type ) {
+			// check that translations are active for this CPT
+			$cpt_slug_is_not_translated = empty( $sitepress_settings['posts_slug_translation']['types'][ $post_type ] );
 
-				$bases['single'] = array_merge( $bases['single'], wp_list_pluck( $slug_translations[1], 'value' ) );
+			if ( $cpt_slug_is_not_translated ) {
+				continue;
 			}
+
+			$slug_translations = WPML_Slug_Translation::get_translations( $post_type );
+
+			if ( ! ( is_array( $slug_translations ) && isset( $slug_translations[1] ) ) ) {
```

Comment:
Wondering, could this be simplified to just `if ( ! isset( $slug_translations[ 1 ] ) ) {}`?